mautic/mautic
# Contact segment validation message fixed

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/2155
| BC breaks? | /
| Deprecations? | /

#### Description:
The contact segment validation displayed message:

> Object(Doctrine\Common\Collections\ArrayCollection): At least one contact segment is required.

The ArrayCollection part should not be there I assume. Also, when I dig deeper to the code, I discovered, that even the second part is not right. It should read:

> Either no list was selected or you do not have access to one of the lists that was selected.

And so this PR changes the error message to show this original message.

I have couple of issues with the current code. There are some oddities.

1. The `app/bundles/LeadBundle/Form/Validator/Constraints/LeadListAccessValidator.php` catches even if the list is empty so I guess that makes calling the second `NotBlank` validator in `app/bundles/EmailBundle/Entity/Email.php` after that one redundant.
2. However, if I change the LeadListAccessValidator to not validate the blank field, the NotBlank validator doesn't trigger. So I rather left it how it is.
3. The error message is ready to take an attribute (`array('%string%' => $l->getName())`), but it is never used in the error message.
4. As I was browsing the Symfony code, many functions we use are marked as deprecated, but I don't see any replacements for them. And the current [Symfony docs about validators](http://symfony.com/doc/current/validation/custom_constraint.html) suggests to use the deprecated methods as well.

#### Steps to test this PR:
1. Re-run the testing steps and the message should change as stated in the PR description.

#### Steps to reproduce the bug:
1. Try to create a new segment email without any list selected.
